files/disable_inactive_freeipa_users.sh

```diff
@@ -69,59 +69,76 @@ else
-        # At least one timestamp is valid.
-        all_timestamps_invalid=false
-
-        # 1. Reformat the timestamp to ISO 8601 format so the date
-        # command can understand it.
-        # 2. Use the date command to convert the ISO 8601 timestamp to
-        # seconds since the epoch.
-        timestamp=${BASH_REMATCH[1]}-${BASH_REMATCH[2]}-${BASH_REMATCH[3]}T${BASH_REMATCH[4]}:${BASH_REMATCH[4]}:${BASH_REMATCH[6]}Z
-        timestamp=$(date --date="$timestamp" +%s)
-
-        # If the timestamp postdates the deadline then the user is
-        # active and should not be disabled.
-        if ((timestamp > disable_deadline)); then
-          disable_user=false
+    # --quiet means no printing unless the p command is used
+    create_timestamp=$(sed --quiet 's/^createTimestamp:\s//p' <<< "$ldapsearch_output")
+
+    # Do we have a valid timestamp in the format that IPA and LDAP use
+    # (YYYYMMDDHHMMSSZ)?
+    if [[ $create_timestamp =~ ^([[:digit:]]{4})([[:digit:]]{2})([[:digit:]]{2})([[:digit:]]{2})([[:digit:]]{2})([[:digit:]]{2})Z$ ]]; then
+      # 1. Reformat the timestamp to ISO 8601 format so the date
+      # command can understand it.
+      # 2. Use the date command to convert the ISO 8601 timestamp to
+      # seconds since the epoch.
+      create_timestamp=${BASH_REMATCH[1]}-${BASH_REMATCH[2]}-${BASH_REMATCH[3]}T${BASH_REMATCH[4]}:${BASH_REMATCH[4]}:${BASH_REMATCH[6]}Z
+      create_timestamp=$(date --date="$create_timestamp" +%s)
+
+      # This if skips any users that were created recently but have
+      # not yet logged in.  We don't want to disable their access yet.
+      if ((create_timestamp < disable_deadline)); then
+        # Extract the user's last authentication timestamp from the
+        # LDAP output.  Note that this timestamp may not exist if the
+        # user has never logged in.
+        #
+        # --quiet means no printing unless the p command is used
+        last_authentication_timestamp=$(sed --quiet 's/^krbLastSuccessfulAuth:\s//p' <<< "$ldapsearch_output")
+
+        # Do we have a valid timestamp in the format that IPA and LDAP
+        # use (YYYYMMDDHHMMSSZ)?
+        if [[ $last_authentication_timestamp =~ ^([[:digit:]]{4})([[:digit:]]{2})([[:digit:]]{2})([[:digit:]]{2})([[:digit:]]{2})([[:digit:]]{2})Z$ ]]; then
+          # 1. Reformat the timestamp to ISO 8601 format so the date
+          # command can understand it.
+          # 2. Use the date command to convert the ISO 8601 timestamp
+          # to seconds since the epoch.
+          last_authentication_timestamp=${BASH_REMATCH[1]}-${BASH_REMATCH[2]}-${BASH_REMATCH[3]}T${BASH_REMATCH[4]}:${BASH_REMATCH[4]}:${BASH_REMATCH[6]}Z
+          last_authentication_timestamp=$(date --date="$last_authentication_timestamp" +%s)
+        else
+          # The timestamp may not exist, but it should never exist but
```

Comment:
```suggestion
          # The timestamp may not exist, but it should never exist and
```

Reply:
I think commit 85a47ce clears things up.  Do you agree?

Reply:
Yep, that works for me.